types/epoch.go

```diff
@@ -48,11 +47,24 @@ func (e *Epoch) String() string {
 		return e.name
 	}
 
-	return hexutil.EncodeBig(e.number)
+	return e.number.String()
+}
+
+// IsEqual check if e equals input
+func (e *Epoch) IsEqual(input *Epoch) bool {
+	if e == nil && input == nil {
+		return true
+	}
+	if e != nil || input != nil {
+		return false
+	}
+	isEqual := e.number.String() == input.number.String() && e.name == input.name
```

Comment:
why not use *big.Int.Cmp method for comparison? it's unnecessary to marshal number to hex for comparison.